Weird error: Something about `for_each` not being defined a lot of different types.

I'm getting a **lot** of errors similar to the following:

```bash
error[E0599]: no method named `for_each` found for struct `bolero::TestTarget<bolero_generator::TypeValueGenerator<activities::append_only_log::AppendOnlyLog>, bolero::test::TestEngine, bolero::BorrowedInput>` in the current scope
   --> bit_network_library/src/activities/append_only_log.rs:224:46
    |
224 |         fuzz!().with_type::<AppendOnlyLog>().for_each(|_| {
    |                                              ^^^^^^^^ method not found in `bolero::TestTarget<bolero_generator::TypeValueGenerator<activities::append_only_log::AppendOnlyLog>, bolero::test::TestEngine, bolero::BorrowedInput>`
    | 
   ::: /home/cfkaran2/.cargo/registry/src/github.com-1ecc6299db9ec823/bolero-0.5.2/src/lib.rs:149:1
    |
149 | pub struct TestTarget<Generator, Engine, InputOwnership> {
    | -------------------------------------------------------- doesn't satisfy `_: std::iter::Iterator`
    | 
   ::: /home/cfkaran2/.cargo/registry/src/github.com-1ecc6299db9ec823/bolero-0.5.2/src/test/mod.rs:14:1
    |
14  | pub struct TestEngine {
    | --------------------- doesn't satisfy `<_ as bolero_engine::Engine<bolero_engine::test::BorrowedGeneratorTest<_, bolero_generator::TypeValueGenerator<activities::append_only_log::AppendOnlyLog>, _>>>::Output = _`
    |
    = note: the method `for_each` exists but the following trait bounds were not satisfied:
            `<bolero::test::TestEngine as bolero_engine::Engine<bolero_engine::test::BorrowedGeneratorTest<_, bolero_generator::TypeValueGenerator<activities::append_only_log::AppendOnlyLog>, _>>>::Output = _`
            `bolero::TestTarget<bolero_generator::TypeValueGenerator<activities::append_only_log::AppendOnlyLog>, bolero::test::TestEngine, bolero::BorrowedInput>: std::iter::Iterator`
            which is required by `&mut bolero::TestTarget<bolero_generator::TypeValueGenerator<activities::append_only_log::AppendOnlyLog>, bolero::test::TestEngine, bolero::BorrowedInput>: std::iter::Iterator`
```

the issue is that I'm implemented `TypeGenerator` on all of these types manually.  I'm concerned that they require additional traits implemented on them; do they?  If so, which ones?

## Meta

```bash
lsb_release -a:
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.5 LTS
Release:	18.04
Codename:	bionic

uname -a:
Linux Hammer 5.4.0-42-generic #46~18.04.1-Ubuntu SMP Fri Jul 10 07:21:24 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

rustc -vV:
rustc 1.45.2 (d3fb005a3 2020-07-31)
binary: rustc
commit-hash: d3fb005a39e62501b8b0b356166e515ae24e2e54
commit-date: 2020-07-31
host: x86_64-unknown-linux-gnu
release: 1.45.2
LLVM version: 10.0

cargo -vV:
cargo 1.45.1 (f242df6ed 2020-07-22)
release: 1.45.1
commit-hash: f242df6edb897f6f69d393a22bb257f5af0f52d0
commit-date: 2020-07-22
```

--- a/bolero/src/lib.rs
+++ b/bolero/src/lib.rs
@@ -367,7 +367,11 @@ cfg_if::cfg_if! {
     }
 }
 
-impl<G, Engine> TestTarget<G, Engine, BorrowedInput> {
+impl<G, Engine> TestTarget<G, Engine, BorrowedInput>
+where
+    G: generator::ValueGenerator,
+    <G as generator::ValueGenerator>::Output: Clone,
+{
     /// Use a cloned value for the test input
     ///
     /// Cloning the test inputs will force a call to [`Clone::clone`]

